src/scheduler.cc

```diff
@@ -543,28 +532,24 @@ bool SchedulerService::is_canonical(ObjRef objref) {
 void SchedulerService::perform_gets() {
   auto get_queue = get_queue_.get();
   // Complete all get tasks that can be completed.
-  for (int i = 0; i < get_queue->size(); ++i) {
-    const std::pair<WorkerId, ObjRef>& get = (*get_queue)[i];
+  get_queue->erase(std::remove_if(get_queue->begin(), get_queue->end(), [&](const std::pair<WorkerId, ObjRef>& get) {
+    ObjStoreId objstoreid = get_store(get.first);
     ObjRef objref = get.second;
-    WorkerId workerid = get.first;
-    ObjStoreId objstoreid = get_store(workerid);
-    if (!has_canonical_objref(objref)) {
+    bool remove = has_canonical_objref(objref);
+    if (remove) {
+      ObjRef canonical_objref = get_canonical_objref(objref);
+      RAY_LOG(RAY_DEBUG, "attempting to get objref " << get.second << " with canonical objref " << canonical_objref << " to objstore " << objstoreid);
+      remove &= (*objtable_.get())[canonical_objref].size() > 0;
+      if (remove) {
+        deliver_object_async_if_necessary(canonical_objref, pick_objstore(canonical_objref), objstoreid);
+        // Notify the relevant objstore about potential aliasing when it's ready
+        alias_notification_queue_.get()->push_back(std::make_pair(objstoreid, std::make_pair(objref, canonical_objref)));
+      }
+    } else {
       RAY_LOG(RAY_ALIAS, "objref " << objref << " does not have a canonical_objref, so continuing");
```

Comment:
I think you're not seeing it here because GitHub collapsed the surrounding space, but this whole thing is now in a lambda. This was actually nontrivial refactoring for improvement -- try expanding it.